pkg/lifecycle/startup/startup.go

```diff
@@ -772,6 +809,27 @@ func autoDetectCIDRBySkipInterface(ifaceRegexes []string, version int) *cnet.IPN
 	return cidr
 }
 
+// autoDetectUsingK8sInternalIP reads K8s Node InternalIP.
+func autoDetectUsingK8sInternalIP(version int, k8sNode *v1.Node) *cnet.IPNet {
```

Comment:
should probably put this in the autodetection package along with the other methods.

Reply:
It would require moving the getLocalCIDR function with it, but I think that's OK